### DISCUSSION TITLE
General Book Layout

### DISCUSSION THREAD

**Discussion by Nickiel12** [ratatui-org/website#2] [Ideas]
For this discussion I would like to talk about how the book will be orders. Right now the plan is roughly:

1. Getting Started
    1. Choose a backend ~~Crossterm/termion/termwiz Backends (tentative, maybe better under reference)~~
    2. Cargo installation
    3. Hello world
3. Tutorial
4. How To (usage example snippets of widgets/layouts
5. References
6. Concepts
    1. Application Patterns (project structure, design patterns, etc)
    2. Threading approaches (tokio/threading/etc)
    3. Backend choice explainations
    4. Other stuff

**Comment by Nickiel12** [comment]
You can find the discord thread for discussion about the book [here](https://discord.com/channels/1070692720437383208/1115053951000268832)

**Reply by joshka** [reply]
Maybe "Choose a backend" would be just a section in installation. We can go into depth on that in the concepts section. That overview looks great :)
